memory/services/SqlServerMemoryStore.cs

```diff
@@ -345,9 +491,9 @@ public async Task<Conversation> GetLastConversationAsync(string userId, int? max
         return conversation;
     }
 
-    public async Task RateLastMessageAsync(string userId, string rating, CancellationToken cancellationToken = default)
+    public Task RateLastMessageAsync(string userId, string rating, CancellationToken cancellationToken = default)
     {
-        await this.ExecuteWithRetryOnTransient(
+        return this.ExecuteWithRetryOnTransient(
```

Comment:
Actually, I would have done this as well, but Nick Chapsas has a video on this and why it is worse for debugging. So the general rule seems to be rather than return a Task if it's doing any real work, you should await it.